### DISCUSSION TITLE
Is consecutive numeric indexing necessary for Graph2Vec?

### DISCUSSION THREAD

**Discussion by bdeng3** [networkx/networkx#5350] [General]
Thanks for the awesome work, networkx is truly helpful when we are dealing with Graph data structure. 

I'm trying to get graph embedding using Graph2vec so that we could compare similarity among graphs. But I'm stuck in this assertion: `assert numeric_indices == node_indices, "The node indexing is wrong."`

Say if we have two graphs, each node in the graph represents a word. We build a mapping so that we could replace text with number. For example, whenever the word "Library" occurs in any graph, we label it with the number "2". In this case, the indexes inside one graph might not be consecutive because the mapping is created from a number of graphs. 

So is it still necessary for enforce consecutive indexing in this case? Or I understand the usage of Graph2Vec wrong?

**Comment by dschult** [comment]
Can you provide a little more code to work from here? A short self-contained example is ideal, showing what does happen and what you'd expect/prefer to happen instead. :}

**Reply by bdeng3** [reply]
Thanks for the quick reply! Really appreciate it. Just realized this assertion comes from `KarateClub` instead of `NetworkX`, I should have open the discussion there. 

It seems that in order to use Graph2Vec algorithm implemented from `KarateClub` we have to created node in the following way:
`a_random_graph.add_node(a_consecutive_numeric_id, "the actual text content")` so that all the nodes inside a graph have consecutive numeric label. It's kinda weird..

**Comment by rossbar** [comment]
Did you intend to open this discussion in NetworkX? AFAICT there is no such assertion anywhere in the NetworkX codebase --- I suspect this discussion was intended for a different project forum.

**Reply by bdeng3** [reply]
Just realized that this assertion comes from the `KarateClub` framework instead of `NetworkX`. My bad..
